extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -846,4 +852,48 @@ export class AmpA4A extends AMP.BaseElement {
       this.stylesheets_.push(linkElem);
     });
   }
+
+  /**
+   * Handle click on links and replace variables in the click URL.
+   * @param {!Event} e
+   */
+  handleClick_(e) {
+    const target = closestByTag(dev().assertElement(e.target), 'A');
+    if (!target || !target.href) {
+      // Not a click on a link.
+      return;
+    }
+    const hrefToExpand =
+    target.getAttribute(ORIGINAL_HREF_ATTRIBUTE) || target.getAttribute('href');
+    if (!hrefToExpand) {
+      return;
+    }
+    const vars = {
+      // The click location assume a switch from ShadowDOM to friendly iframe
+      // based rendering, in which case the document relative positions are
+      // really ad-relative.
+      'CLICK_X': () => {
+        return e.pageX;
+      },
+      'CLICK_Y': () => {
+        return e.pageY;
+      },
+    };
+    const newHref = urlReplacementsFor(this.win).expandSync(
+        hrefToExpand, vars, undefined, /* opt_whitelist */ {
+          // For now we only allow to replace the click location vars
+          // and nothing else.
+          'CLICK_X': true,
```

Comment:
Yeah, sigh. I was considering an array, since O(small N) is likely faster than more complicated O(1), but this is slightly more idiomatic.